functional/FunctionalTest.java

```diff
@@ -1074,11 +1074,17 @@ public static void copyObject_test7() throws Exception {
 
     copyConditions.setUnmodified(dateRepresentation);
 
+    ErrorResponseException expectedException = null;
+
     try {
       client.copyObject(bucketName, fileName, secondBucketName, copyConditions);
     } catch (ErrorResponseException e) {
       // File should not be copied as object was modified after date set in copyConditions.
-      ignore();
+      expectedException = e;
+    }
+
+    if (expectedException == null || !expectedException.errorResponse().code().equals("PreconditionFailed")) {
```

Comment:
The actual bug is having `client.removeObject(destBucketName, filename);` in `copyObject_test7()`.  If you remove that code, there is no special exception handling required.